pkg/steps/optional_operators.go

```diff
@@ -0,0 +1,107 @@
+package steps
+
+import (
+	"fmt"
+	"strings"
+
+	coreapi "k8s.io/api/core/v1"
+)
+
+const (
+	ooIndex            = "OO_INDEX"
+	ooPackage          = "OO_PACKAGE"
+	ooChannel          = "OO_CHANNEL"
+	ooInstallNamespace = "OO_INSTALL_NAMESPACE"
+	ooTargetNamespaces = "OO_TARGET_NAMESPACES"
+)
+
+// optionalOperator is the information needed by the optional operator installation
+// steps to be able to subscribe and install an optional operator
+type optionalOperator struct {
+	// The pullspec of an index image
+	Index string
+	// The name of the operator package to be installed
+	Pkg string
+	// The name of the operator channel to track
+	Channel string
+	// The namespace into which the operator and catalog will be installed (optional)
+	Namespace string
+	// A comma-separated list of namespaces the operator will target. If empty,
+	// all namespaces will be targeted.
+	TargetNamespaces []string
+}
+
+// getter is a subset of api.Parameters
+type getter interface {
+	Get(name string) (string, error)
+}
+
+// resolveOptionalOperator extracts an optionalOperator instance from the
+// given parameters
+func resolveOptionalOperator(parameters getter) (*optionalOperator, error) {
+	var err error
+	var oo optionalOperator
+
+	if oo.Index, err = parameters.Get(ooIndex); err != nil {
```

Comment:
```go
for param, valuePointer := range map[string]*string{
  ooIndex: &oo.Index,
  ooPackage: &oo.Pkg,
  ooChannel: &oo.Channel,
} {
    if value, err := parameters.Get(parameter); err != nil {
        return nil, err
    }
    valuePointer = value
}
```

?

Reply:
This one feels like borderline too smart for what it really solves, but after some deliberation I did it your way :)